File: app/src/main/java/com/zaclimon/aceiptv/ui/auth/AuthPresenter.java
package com.zaclimon.aceiptv.ui.auth;

import com.zaclimon.aceiptv.repository.SharedPreferencesRepository;

/**
 * Interface that defines a valid Presenter used for the authentication process.
 *
 * @author zaclimon
 * Creation date: 07/06/17
 */

public interface AuthPresenter {

    /**
     * Validates the credentials sent by a user and calls, if found, required callback methods.
     * @param username the user's username
     * @param password the user's password
     * @param sharedPreferencesRepository the repository of SharedPreferences.
     */
    void validateInfo(String username, String password, SharedPreferencesRepository sharedPreferencesRepository);
}

File: app/src/main/java/com/zaclimon/tsutaeru/ui/auth/AuthPresenter.kt
package com.zaclimon.tsutaeru.ui.auth

import com.zaclimon.tsutaeru.repository.SharedPreferencesRepository

/**
 * Interface that defines a valid Presenter used for the authentication process.
 *
 * @author zaclimon
 */
interface AuthPresenter {

    /**
     * Validates the credentials sent by a user and calls, if found, required callback methods.
     *
     * @param url                         the user's provider url
     * @param sharedPreferencesRepository the repository of SharedPreferences.
     */
    fun validateInfo(url: String, sharedPreferencesRepository: SharedPreferencesRepository)

}